tern/__main__.py

```diff
@@ -69,6 +69,33 @@ def create_top_dir(working_dir=None):
         os.makedirs(top_dir)
 
 
+def execute_image(args):
+    ''' Executes container images using the given inputs '''
+    if args.type == "docker":
+        # Check if the image is of image:tag
+        # or image@digest_type:digest format
+        if not check_image_string(args.image):
+            sys.stderr.write('Error running Tern\n'
+                             'Please provide docker image '
+                             'string in image:tag or '
+                             'image@digest_type:digest format\n')
+            sys.exit(1)
+        if general.check_tar(args.image):
+            logger.error("%s", errors.incorrect_raw_option)
```

Comment:
I think this option should be different from the `-i` option now. Or else this is going to get more complicated. @rnjudge What do you think?

Reply:
Not sure I understand your question. The `-w` option is already separate from the `-i` option, right?

Reply:
Shouldn't our check here then be `if args.raw`?

Reply:
I believe this check here is to make sure that the user didn't provide a raw option tarball when they are trying to provide a docker or OCI image.

Reply:
@nishakm, this check is part of the `execute_image` method and this method is only for docker or OCI image execution. The `args.raw_image` still I am keeping the part `do_main` method.

Reply:
Confirmed that this is just a validation check to make sure the user didn't accidentally choose the `-i` option but provide a raw image tarball. Seems OK to me.